src/axom/primal/operators/detail/in_curved_polygon_impl.hpp

```diff
@@ -25,82 +25,119 @@ namespace primal
+    return 0;
 
-  double dotprod = axom::utilities::clampVal(V1.dot(V2), -1.0, 1.0);
+  // Compute signed angle between vectors
+  double dotprod = axom::utilities::clampVal(
+    Vector<T, 2>::dot_product(V1.unitVector(), V2.unitVector()),
+    -1.0,
+    1.0);
 
   return -0.5 * M_1_PI * acos(dotprod) * ((orient > 0) ? 1 : -1);
 }
 
 /*!
- * \brief Directly compute the winding number at an endpoint of a 
+ * \brief Directly compute the winding number at either endpoint of a 
  *        Bezier curve with a convex control polygon
  *
  * \param [in] is_init Boolean value indicating endpoint is at t=0
  * \param [in] c The BezierCurve object to compute the winding number along
+ * \param [in] edge_tol The physical distance level which objects are indistinguishable
+ * \param [in] EPS Miscellaneous numerical tolerance for isNearlyZero
+ * \pre Control polygon for c must be convex
  *
  * The winding number for a Bezier curve with a convex control polygon is
  * given by the signed angle between the tangent vector at that endpoint and
- * the vector in the direction of the other endpoint. The query can be located
- * on both endpoints if it is closed.
+ * the vector in the direction of the other endpoint. 
+ * 
+ * The query can be located on both endpoints if it is closed, in which case
+ * the angle is that between the tangent lines at both endpoints
  * 
  * \return 
  */
 template <typename T>
-double convex_endpoint_winding_number(bool is_init, const BezierCurve<T, 2>& c)
+double convex_endpoint_winding_number(const Point<T, 2>& q,
+                                      const BezierCurve<T, 2>& c,
+                                      double edge_tol,
+                                      double EPS)
 {
   const int ord = c.getOrder();
 
   if(ord == 1) return 0;
 
-  Vector<T, 2> V1, V2;
-  if(is_init)  // Query point is on initial endpoint
-  {
-    V1 = Vector<T, 2>(c[0], c[1]).unitVector();
-    V2 = Vector<T, 2>(c[0], c[ord]).unitVector();
-  }
-  else  // Query point is on terminal endpoint
-  {
-    V1 = Vector<T, 2>(c[ord], c[0]).unitVector();
-    V2 = Vector<T, 2>(c[ord], c[ord - 1]).unitVector();
-  }
+  int idx;
+
+  // Need to find vectors that subtend the entire curve.
+  //   We must ignore duplicate nodes
+  for(idx = 0; idx <= ord; ++idx)
+    if(squared_distance(q, c[idx]) > edge_tol * edge_tol) break;
+  Vector<T, 2> V1(q, c[idx]);
+
+  for(idx = ord; idx >= 0; --idx)
+    if(squared_distance(q, c[idx]) > edge_tol * edge_tol) break;
+  Vector<T, 2> V2(q, c[idx]);
 
   // clang-format off
   double orient = axom::numerics::determinant(V1[0] - V2[0], V2[0], 
                                               V1[1] - V2[1], V2[1]);
   // clang-format on
 
-  // If the tangent lines are parallel, winding number is ambiguous
-  if(axom::utilities::isNearlyEqual(orient, 0.0)) return 0;
-
-  double dotprod = axom::utilities::clampVal(V1.dot(V2), -1.0, 1.0);
+  // This means the bounding vectors are anti-parallel.
```

Comment:
I think it would depend on whether `ord` is more likely to be greater than or less than 3, since the cost of the order check and the nonzero check I would think are pretty similar. I'll add it in there on the assumption that we would be doing this on cubics most often. If that isn't the case, I don't think it's worth it to do the order check, as we still have to do the nonzero check in the other cases.